src/SteadyHeap.h

```diff
@@ -0,0 +1,110 @@
+#ifndef _STEADY_HEAP_H_
+#define _STEADY_HEAP_H_
+
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+namespace pipert {
+
+/// A SteadyHeap wraps the functionality of a heap that can be restrained to
+/// occupy a fix-sized space in memory. It does this by encapsulating a vector
+/// and by providing only the interface to reserve the maximum amount of memory
+/// this heap is allowed to consume, as well as providing methods for handling
+/// its contents as a heap. This container wrapper is specially adapted to our
+/// needs, and contains only the functions the library uses. It does not give a
+/// complete interface for a generic heap, and the implementation is expected to
+/// be extended when new requirements arise in the library.
+///
+/// \tparam T The type of data the heap will contain as its elements.
+/// \tparam StrictOrdering The type of the comparison operation that will be
+/// used to determine the positioning of elements in the heap. It should satisfy
+/// the requirements of _Compare_, meaning that the return value of the
+/// comparison function, when contextually converted to `bool`, should yield
+/// `true` if its first argument appears before the second in the strict weak
+/// ordering induced by the type of its parameters. Comparison using less-than
+/// yields a max heap, and greater-than yields a min heap.
+template <typename T, typename StrictOrdering>
+class SteadyHeap {
+ public:
+  /// Construct a SteadyHeap with a strict ordering.
+  ///
+  /// \param ordering See the template parameter StrictOrdering for more
+  /// information. Probably best implemented as a std::function wrapper.
+  SteadyHeap(StrictOrdering ordering) : ordering_(ordering) {}
+
+  /// \return Whether the container is empty.
```

Comment:
I am always wondering that a simple return would be enough. But if you only put this, nothing will be displayed in the briefing in the class listing for this. Sometimes i skip the return just to have a briefing, but sometimes I give both despite the redundancy...

Reply:
Yes, it's perplexing. Here, I thought that omitting the briefing for something that essentially mirrors the STL isn't really a problem. Seems like there's no way to avoid duplication, as writing a briefing only containing a sentence that starts with "return" also feels weird. In my opinion, it may be best to avoid bloating the in-code documentation, and just have one or the other.

Anyways, seeing that all changes are approved, I'll try auto-merging the branch now. Thank you for your review.